database/db.go

```diff
@@ -21,6 +21,7 @@ import (
 	"go.uber.org/zap/zapcore"
 	"golang.org/x/sync/errgroup"
 	"golang.org/x/sync/semaphore"
+	_ "modernc.org/sqlite"
```

Comment:
While in theory having a CGo-free SQLite database driver sounds tempting, this library introduces a hard limitation on targeted platforms. Please see the [README section](https://pkg.go.dev/modernc.org/sqlite#hdr-Supported_platforms_and_architectures) or this [random issue](https://gitlab.com/cznic/sqlite/-/issues/88). Since this code is introduced in our `database` package, I am unsure how this applies to downstream consumers such as Icinga DB.

So even if CGo is a nuisance, I would recommend using the good old [`github.com/mattn/go-sqlite3` library](https://github.com/mattn/go-sqlite3), as we already do in Icinga DB for the `icingadb-migrate` utility.

Reply:
No big deal. Honestly, it's not that obvious. In the past, I would have liked to use this library as well.